binary/type.proto

```diff
@@ -275,22 +137,17 @@ message Type {
 
     message TimestampMicroTZ {
         string timezone = 1;
-
-        PhysicalType physical_type = 2;
-
-        enum PhysicalType {
-            SYSTEM_DEFAULT = 0;
-        }
+        Variation variation = 2;
     }
 
     message TimestampTZ {
         string timezone = 1;
+        Variation variation = 2;
+    }
 
-        PhysicalType physical_type = 2;
-
-        enum PhysicalType {
-            SYSTEM_DEFAULT = 0;
-        }
+    message Variation {
```

Comment:
A possible more natural way of modeling this in protobuf is to use have different physical type systems define [extensions](https://developers.google.com/protocol-buffers/docs/proto#extensions).  Did you consider that as an alternative? (I don't think this needs to hold up the PR).

Reply:
I don't really see much benefit to the extensions approach in this case. Generally, it is an interesting question whether to expose some objects with extensions but I generally feel like any kind of pure black box that an extension would provide would weaken the spec. In the pattern here the spec still has information about how the extended functionality works, even if declared beyond the core project.

Reply:
Extensions provide the ability to more strongly model variations.  There would still be two classes:
1.  In spec (those checked into this repo) and an extension range would be reserved for these).
2. Those not checked into the repo (blackbox) and would live in a separate range.

I was thinking have one extension for each system type. e.g. `ArrowVarationExtension` or `AvroVariationExtension`.  At least for Arrow variations seem like they might stack (largestring and dictionary encoding).

Reply:
It doesn't really seem like extensions provide a stronger model. For example, when trying to decide whether a particular function can be bound to a particular type:

- With the approach provided in the PR, the rules about binding are clear. If the variation inherits from logical, the functions can be bound to any function declared against the base logical type. It also fully defines what a complete type definition is.
- If using extensions, when trying to bind, can you bind to a function definition for the base logical type against a type that has one or more extensions properties defined?

I also haven't seen extensions successfully in a project in this way (it seems like they are only a bucket for arbitrary as opposed to structured information).